ui/src/EventForm.tsx

```diff
@@ -0,0 +1,129 @@
+import React, { useState } from "react";
+import { formatDate } from "./lib";
+import s from "./App.module.css";
+
+interface FormProps {
+  initialStartDate: string;
+  initialEndDate: string;
+  initialStartTime: string;
+  initialEndTime: string;
+  initialTitle: string;
+  onSave: Function;
+  isCreate: boolean;
+}
+
+const EventForm = ({
+  initialStartDate,
+  initialEndDate,
+  initialStartTime,
+  initialEndTime,
+  initialTitle,
+  onSave,
+  isCreate,
+}: FormProps) => {
+  const [startDate, setStartDate] = useState<string>(initialStartDate);
+  const [endDate, setEndDate] = useState<string>(initialEndDate);
+  const [startTime, setStartTime] = useState<string>(initialStartTime);
+  const [endTime, setEndTime] = useState<string>(initialEndTime);
+  const [error, setError] = useState<string | null>(null);
+  const [title, setTitle] = useState<string>(initialTitle);
+
+  const handleSave = async (_: React.MouseEvent<HTMLButtonElement>) => {
+    const startDateAndTime: string = `${startDate}T${startTime}`;
+    const endDateAndTime: string = `${endDate}T${endTime}`;
```

Comment:
Non-blocking, but this sort of string interpolation could also be nice to have in a helper function since it's used 8 times in our code already, would help keep the code less brittle